### PR TITLE
Add missing localisation for settings enums

### PR DESCRIPTION
**Pull request by its5Q**
I've addressed all the comments in #19623. This PR adds localisation support for most dropdown settings.

### CODE REVIEW

**Comment by its5Q** [comment]
It's failing on master branch, something is wrong on master branch

**Comment by its5Q** [comment]
This is strange, it passes and fails on almost the exact same codebase, the only difference was changing enum member name from "JPG" to "Jpg".
https://github.com/ppy/osu/actions/runs/2817692398
https://github.com/ppy/osu/actions/runs/2817759916

File: osu.Game/Localisation/GameplaySettingsStrings.cs [hunk 1]
```diff
@@ -104,6 +104,31 @@ public static class GameplaySettingsStrings
...
 
+        /// <summary>
+        /// "Hide during gameplay"
+        /// </summary>
+        public static LocalisableString HideDuringGameplay => new TranslatableString(getKey(@"hide_during_gameplay"), @"Hide during gameplay");
+
+        /// <summary>
+        /// "Always"
+        /// </summary>
+        public static LocalisableString Always => new TranslatableString(getKey(@"always"), @"Always");
```
Comment (smoogipoo) [inline osu.Game/Localisation/GameplaySettingsStrings.cs:115]: There's a few duplicate strings here that should be moved into `CommonStrings` imo. Thoughts @peppy ?

Comment (peppy) [reply osu.Game/Localisation/GameplaySettingsStrings.cs:115]: I intentionally asked them to move these into specific implementations, as they will translate differently in some languages, such as japanese, for certain settings.

They do need proper prefixes in the keys though. This should be "SomethingAlways" not just "Always".

Comment (its5Q) [reply osu.Game/Localisation/GameplaySettingsStrings.cs]: I added prefixes to common strings to differenciate between them.

File: osu.Game/Localisation/GameplaySettingsStrings.cs [hunk 2]
```diff
@@ -104,6 +104,31 @@ public static class GameplaySettingsStrings
...
+
+        /// <summary>
+        /// "Always"
+        /// </summary>
+        public static LocalisableString AlwaysShowHUD => new TranslatableString(getKey(@"always_show_hud"), @"Always");
+
+        /// <summary>
+        /// "Never"
+        /// </summary>
+        public static LocalisableString NeverShowHUD => new TranslatableString(getKey(@"never"), @"Never");
```
Comment (smoogipoo) [inline osu.Game/Localisation/GameplaySettingsStrings.cs:120]:
```suggestion
        public static LocalisableString NeverShowHUD => new TranslatableString(getKey(@"never_show_hud"), @"Never");
```

File: osu.Game/Localisation/LayoutSettingsStrings.cs [hunk 3]
```diff
@@ -29,6 +29,26 @@ public static class LayoutSettingsStrings
         /// </summary>
         public static LocalisableString FullscreenMacOSNote => new TranslatableString(getKey(@"fullscreen_macos_note"), @"Using fullscreen on macOS makes interacting with the menu bar and spaces no longer work, and may lead to freezes if a system dialog is presented. Using borderless is recommended.");
 
+        /// <summary>
+        /// "Excluding overlays"
+        /// </summary>
+        public static LocalisableString ExcludingOverlays => new TranslatableString(getKey(@"excluding_overlays"), @"Excluding overlays");
```
Comment (smoogipoo) [inline osu.Game/Localisation/LayoutSettingsStrings.cs:35]: Maybe something like `ScaleEverythingExcludingOverlays`?

File: osu.Game/Localisation/LayoutSettingsStrings.cs [hunk 4]
```diff
@@ -29,6 +29,26 @@ public static class LayoutSettingsStrings
...
+
+        /// <summary>
+        /// "Everything"
+        /// </summary>
+        public static LocalisableString ScaleEverything => new TranslatableString(getKey(@"everything"), @"Everything");
```
Comment (smoogipoo) [inline osu.Game/Localisation/LayoutSettingsStrings.cs:40]:
```suggestion
        public static LocalisableString ScaleEverything => new TranslatableString(getKey(@"scale_everything"), @"Everything");
```

File: osu.Game/Localisation/LayoutSettingsStrings.cs [hunk 5]
```diff
@@ -29,6 +29,26 @@ public static class LayoutSettingsStrings
...
+
+        /// <summary>
+        /// "Gameplay"
+        /// </summary>
+        public static LocalisableString ScaleGameplay => new TranslatableString(getKey(@"gameplay"), @"Gameplay");
```
Comment (smoogipoo) [inline osu.Game/Localisation/LayoutSettingsStrings.cs:45]:
```suggestion
        public static LocalisableString ScaleGameplay => new TranslatableString(getKey(@"scale_gameplay"), @"Gameplay");
```

File: osu.Game/Localisation/LayoutSettingsStrings.cs [hunk 6]
```diff
@@ -29,6 +29,26 @@ public static class LayoutSettingsStrings
...
+
+        /// <summary>
+        /// "Off"
+        /// </summary>
+        public static LocalisableString ScalingOff => new TranslatableString(getKey(@"off"), @"Off");
```
Comment (smoogipoo) [inline osu.Game/Localisation/LayoutSettingsStrings.cs:50]:
```suggestion
        public static LocalisableString ScalingOff => new TranslatableString(getKey(@"scaling_off"), @"Off");
```

File: osu.Game/Localisation/MouseSettingsStrings.cs [hunk 7]
```diff
@@ -64,6 +64,21 @@ public static class MouseSettingsStrings
         /// </summary>
         public static LocalisableString HighPrecisionPlatformWarning => new TranslatableString(getKey(@"high_precision_platform_warning"), @"This setting has known issues on your platform. If you encounter problems, it is recommended to adjust sensitivity externally and keep this disabled for now.");
 
+        /// <summary>
+        /// "Always"
+        /// </summary>
+        public static LocalisableString AlwaysConfine => new TranslatableString(getKey(@"always"), @"Always");
```
Comment (smoogipoo) [inline osu.Game/Localisation/MouseSettingsStrings.cs:70]:
```suggestion
        public static LocalisableString AlwaysConfine => new TranslatableString(getKey(@"always_confine"), @"Always");
```

File: osu.Game/Localisation/MouseSettingsStrings.cs [hunk 8]
```diff
@@ -64,6 +64,21 @@ public static class MouseSettingsStrings
...
+
+        /// <summary>
+        /// "During Gameplay"
+        /// </summary>
+        public static LocalisableString ConfineDuringGameplay => new TranslatableString(getKey(@"during_gameplay"), @"During Gameplay");
```
Comment (smoogipoo) [inline osu.Game/Localisation/MouseSettingsStrings.cs:75]:
```suggestion
        public static LocalisableString ConfineDuringGameplay => new TranslatableString(getKey(@"confine_during_gameplay"), @"During Gameplay");
```

File: osu.Game/Localisation/MouseSettingsStrings.cs [hunk 9]
```diff
@@ -64,6 +64,21 @@ public static class MouseSettingsStrings
...
+
+        /// <summary>
+        /// "Never"
+        /// </summary>
+        public static LocalisableString NeverConfine => new TranslatableString(getKey(@"never"), @"Never");
```
Comment (smoogipoo) [inline osu.Game/Localisation/MouseSettingsStrings.cs:80]:
```suggestion
        public static LocalisableString NeverConfine => new TranslatableString(getKey(@"never_confine"), @"Never");
```

File: osu.Game/Localisation/OnlineSettingsStrings.cs [hunk 10]
```diff
@@ -64,6 +64,21 @@ public static class OnlineSettingsStrings
...
 
+        /// <summary>
+        /// "Hide identifiable information"
+        /// </summary>
+        public static LocalisableString HideIdentifiableInformation => new TranslatableString(getKey(@"hide_identifiable_information"), @"Hide identifiable information");
+
+        /// <summary>
+        /// "Full"
+        /// </summary>
+        public static LocalisableString DiscordPresenceFull => new TranslatableString(getKey(@"full"), @"Full");
```
Comment (smoogipoo) [inline osu.Game/Localisation/OnlineSettingsStrings.cs:75]:
```suggestion
        public static LocalisableString DiscordPresenceFull => new TranslatableString(getKey(@"discord_presence_full"), @"Full");
```

File: osu.Game/Localisation/OnlineSettingsStrings.cs [hunk 11]
```diff
@@ -64,6 +64,21 @@ public static class OnlineSettingsStrings
...
+
+        /// <summary>
+        /// "Off"
+        /// </summary>
+        public static LocalisableString DiscordPresenceOff => new TranslatableString(getKey(@"off"), @"Off");
```
Comment (smoogipoo) [inline osu.Game/Localisation/OnlineSettingsStrings.cs:80]:
```suggestion
        public static LocalisableString DiscordPresenceOff => new TranslatableString(getKey(@"discord_presence_off"), @"Off");
```

File: osu.Game/Localisation/RulesetSettingsStrings.cs [hunk 12]
```diff
@@ -14,6 +14,21 @@ public static class RulesetSettingsStrings
         /// </summary>
         public static LocalisableString Rulesets => new TranslatableString(getKey(@"rulesets"), @"Rulesets");
 
+        /// <summary>
+        /// "None"
+        /// </summary>
+        public static LocalisableString BorderNone => new TranslatableString(getKey(@"none"), @"None");
```
Comment (smoogipoo) [inline osu.Game/Localisation/RulesetSettingsStrings.cs:20]:
```suggestion
        public static LocalisableString BorderNone => new TranslatableString(getKey(@"no_borders"), @"None");
```

File: osu.Game/Localisation/RulesetSettingsStrings.cs [hunk 13]
```diff
@@ -14,6 +14,21 @@ public static class RulesetSettingsStrings
...
+
+        /// <summary>
+        /// "Corners"
+        /// </summary>
+        public static LocalisableString BorderCorners => new TranslatableString(getKey(@"corners"), @"Corners");
```
Comment (smoogipoo) [inline osu.Game/Localisation/RulesetSettingsStrings.cs:25]:
```suggestion
        public static LocalisableString BorderCorners => new TranslatableString(getKey(@"corner_borders"), @"Corners");
```

File: osu.Game/Localisation/RulesetSettingsStrings.cs [hunk 14]
```diff
@@ -14,6 +14,21 @@ public static class RulesetSettingsStrings
...
+
+        /// <summary>
+        /// "Full"
+        /// </summary>
+        public static LocalisableString BorderFull => new TranslatableString(getKey(@"full"), @"Full");
```
Comment (smoogipoo) [inline osu.Game/Localisation/RulesetSettingsStrings.cs:30]:
```suggestion
        public static LocalisableString BorderFull => new TranslatableString(getKey(@"full_borders"), @"Full");
```

File: osu.Game/Localisation/UserInterfaceStrings.cs [hunk 15]
```diff
@@ -114,6 +114,46 @@ public static class UserInterfaceStrings
...
 
+        /// <summary>
+        /// "Beatmap (with storyboard / video)"
+        /// </summary>
+        public static LocalisableString BeatmapWithStoryboard => new TranslatableString(getKey(@"beatmap_with_storyboard"), @"Beatmap (with storyboard / video)");
+
+        /// <summary>
+        /// "Always"
+        /// </summary>
+        public static LocalisableString AlwaysSeasonalBackground => new TranslatableString(getKey(@"always"), @"Always");
```
Comment (smoogipoo) [inline osu.Game/Localisation/UserInterfaceStrings.cs:125]:
```suggestion
        public static LocalisableString AlwaysSeasonalBackground => new TranslatableString(getKey(@"always_seasonal_backgrounds"), @"Always");
```

File: osu.Game/Localisation/UserInterfaceStrings.cs [hunk 16]
```diff
@@ -114,6 +114,46 @@ public static class UserInterfaceStrings
...
+
+        /// <summary>
+        /// "Never"
+        /// </summary>
+        public static LocalisableString NeverSeasonalBackground => new TranslatableString(getKey(@"never"), @"Never");
```
Comment (smoogipoo) [inline osu.Game/Localisation/UserInterfaceStrings.cs:130]:
```suggestion
        public static LocalisableString NeverSeasonalBackground => new TranslatableString(getKey(@"never_seasonal_backgrounds"), @"Never");
```

File: osu.Game/Localisation/UserInterfaceStrings.cs [hunk 17]
```diff
@@ -114,6 +114,46 @@ public static class UserInterfaceStrings
...
+
+        /// <summary>
+        /// "Sometimes"
+        /// </summary>
+        public static LocalisableString SometimesSeasonalBackground => new TranslatableString(getKey(@"sometimes"), @"Sometimes");
```
Comment (smoogipoo) [inline osu.Game/Localisation/UserInterfaceStrings.cs:135]:
```suggestion
        public static LocalisableString SometimesSeasonalBackground => new TranslatableString(getKey(@"sometimes_seasonal_backgrounds"), @"Sometimes");
```

File: osu.Game/Localisation/UserInterfaceStrings.cs [hunk 18]
```diff
@@ -114,6 +114,46 @@ public static class UserInterfaceStrings
...
+
+        /// <summary>
+        /// "Sequential"
+        /// </summary>
+        public static LocalisableString SequentialHotkeyStyle => new TranslatableString(getKey(@"sequential"), @"Sequential");
+
+        /// <summary>
+        /// "Classic"
+        /// </summary>
+        public static LocalisableString ClassicHotkeyStyle => new TranslatableString(getKey(@"classic"), @"Classic");
```
Comment (smoogipoo) [inline osu.Game/Localisation/UserInterfaceStrings.cs:145]:
```suggestion
        public static LocalisableString ClassicHotkeyStyle => new TranslatableString(getKey(@"mods_classic_hotkeys"), @"Classic");
```

File: osu.Game/Localisation/UserInterfaceStrings.cs [hunk 19]
```diff
@@ -114,6 +114,46 @@ public static class UserInterfaceStrings
...
+
+        /// <summary>
+        /// "Sometimes"
+        /// </summary>
+        public static LocalisableString SometimesSeasonalBackground => new TranslatableString(getKey(@"sometimes"), @"Sometimes");
+
+        /// <summary>
+        /// "Sequential"
+        /// </summary>
+        public static LocalisableString SequentialHotkeyStyle => new TranslatableString(getKey(@"sequential"), @"Sequential");
```
Comment (smoogipoo) [inline osu.Game/Localisation/UserInterfaceStrings.cs:140]:
```suggestion
        public static LocalisableString SequentialHotkeyStyle => new TranslatableString(getKey(@"mods_sequential_hotkeys"), @"Sequential");
```

**Review by smoogipoo — CHANGES_REQUESTED** [review]

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**